### PR TITLE
[string] Minor cleanup

### PR DESCRIPTION
**Pull request by milseman**
Just some cleaning in preparation for smol strings.

### CODE REVIEW

**Comment by milseman** [comment]
@swift-ci please test

**Comment by milseman** [comment]
@swift-ci please benchmark

File: stdlib/public/core/StringObject.swift [hunk 1]
```diff
@@ -694,34 +686,39 @@ extension _StringObject {
...
+      _sanityCheck(isSmall == (isValue && isSmallOrObjC))
+      _sanityCheck(isUnmanaged == (isValue && !isSmallOrObjC))
+      _sanityCheck(isCocoa == (!isValue && isSmallOrObjC))
+      _sanityCheck(isNative == (!isValue && !isSmallOrObjC))
+    }
+#endif
+
 #if arch(i386) || arch(arm)
-    var variant: _Variant
-    var bits: UInt
```
Comment (lorentey) [inline stdlib/public/core/StringObject.swift]: Oops 🤦‍♂️

**Review by lorentey — APPROVED** [review]
Nice!

File: stdlib/public/core/StringLegacy.swift [hunk 2]
```diff
@@ -48,23 +48,22 @@ extension String {
...
-    if count == 0 {
-      self = ""
-    } else if count == 1 {
-      self = repeatedValue
+    guard count > 1 else {
+      self = count == 0 ? "" : repeatedValue
+      return
+    }
+
+    precondition(count > 0, "Negative count not allowed")
```
Comment (otaviolima) [inline stdlib/public/core/StringLegacy.swift]: `precondition` here is unnecessary due the `guard` on line 51

**Pull request closed without merging**